appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/cdi/interceptor/AbstractInterceptor.java

```diff
@@ -96,10 +101,13 @@ private Object timeoutInvocation(InvocationContext context) throws Exception {
     private <E extends Member & AnnotatedElement> Object preInterceptor(InvocationContext context, E element) throws Exception {
         initService();
         if (metricsService.isEnabled()) {
+            //FIXME there is an issue here: the element does not correctly reflect the updated annotations
```

Comment:
I opted to not open that box because of the change of behaviour it might bring. I don't think it is much work but it is this ugly inconsistency in CDI where you cannot ask for the `Bean` that is intercepted. Weld has a non standard way around it which allows to access the CDI level abstraction which has the correct picture. To be clear: this is just the difference between dynamic adds/removes of annotations on CDI level considered or not.